src/components/Widget.tsx

```diff
@@ -131,9 +131,9 @@ export function TestableWidget(props: PropsWithChildren<TestableWidgetProps>) {
           <I18nProvider locale={locale}>
             <DialogWrapper ref={setDialog} />
             <DialogProvider value={props.dialog || dialog}>
-              <ErrorBoundary onError={props.onError}>
-                <ReduxProvider store={store}>
-                  <AtomProvider initialValues={props.initialAtomValues}>
+              <ReduxProvider store={store}>
+                <AtomProvider initialValues={props.initialAtomValues}>
+                  <ErrorBoundary onError={props.onError}>
```

Comment:
This may be dangerous - if there is a problem with the atoms, it may be out of scope of the ErrorBoundary.

It is safer to implement the error state (`src/state/error.ts`) using `Context`, so that it can be independent and the `ErrorBoundary` can still be at a higher level than `Redux` and `Atom` providers.